### DISCUSSION TITLE
How to enable authorization in entgql?

### DISCUSSION THREAD

**Discussion by gemone** [ent/ent#3729] [Q&A]
Hello,

Can fields be authenticated in entgql? And reuse my existing JWT authentication? Just like https://graphql.org/learn/authorization/.

I see that [privacy](https://entgo.io/docs/privacy/) can control database queries. Can user information be passed through context.Context in entgql?

Thanks!

**Comment by crossworth** [comment]
Hello @gemone, what I have done on past projects is having a global middleware that store the user on context, then I define a custom GraphQL directive that check for authenticated users (the directive only check if there are a user on the context, nothing fancy, since the JWT middleware already validated the token if present) on the fields that needed to be authenticated to be used.

But see that all the authorization is outside Ent, so you can use different methods.

You can add directives with [entgql Directives annotation](https://pkg.go.dev/entgo.io/contrib/entgql#Directives) and implement it using [gqlgen  Schema Directives](https://gqlgen.com/reference/directives/).

**Reply by gemone** [reply]
Thanks @crossworth ，I got it.
